server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/message/view/MessageViewFactory.java

```diff
@@ -44,96 +55,166 @@
 import com.google.common.collect.Multimaps;
 
 public interface MessageViewFactory<T extends MessageView> {
+
+    Logger LOGGER = LoggerFactory.getLogger(MessageViewFactory.class);
+
     KeywordsCombiner KEYWORDS_COMBINER = new KeywordsCombiner();
     Keywords.KeywordsFactory KEYWORDS_FACTORY = Keywords.lenientFactory();
     String JMAP_MULTIVALUED_FIELD_DELIMITER = "\n";
 
-    T fromMessageResults(Collection<MessageResult> messageResults) throws MailboxException;
+    List<T> fromMessageIds(List<MessageId> messageIds, MailboxSession mailboxSession) throws MailboxException;
```

Comment:
the session is created by the `AuthenticatedRequest.getMailboxSession()`, I decided to pass that far created object to the factory rather than create it inside the factory because that's not easy to know the username